### PR TITLE
Feature/14780 implement auth token

### PR DESCRIPTION
**Pull request by jon-shipley**
JWT

Add JWT to the questions API
Store the JWT the pupil-spa
New jwtService in the admin app that can verify the incoming token

### CODE REVIEW

File: admin/spec/lib/validator/hdf-validation-spec.js [hunk 1]
```diff
@@ -16,9 +16,9 @@ describe('HDF validator', function () {
     }
   }
 
-  function notAllowed () {
-    return [':', ';', ',', '.', '!', '@', '#', '$', ' ', '%', '^', '&', '*', '(', ')', '_', '+', '=']
-  }
+  // function notAllowed () {
```
Comment (GeorgeChatzigiannis) [inline admin/spec/lib/validator/hdf-validation-spec.js]: do you need these comments?

File: admin/services/jwt-service.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      iss: 'MTC Admin',                                       // Issuer
+      sub: pupil._id,                                         // Subject
+      exp: Math.floor(Date.now() / 1000) + (60 * 60),         // Expiry
+      nbf: Math.floor(Date.now() / 1000),                     // Not before
+      jwi: jwtId                                              // JWT token ID
+    }
+
+    // Construct a JWT token
+    const token = await jwt.sign(payload, pupil.jwtSecret)
+    return token
```
Comment (GeorgeChatzigiannis) [inline admin/services/jwt-service.js]: you don't need the variable here you can just return await jwt.sign(...

File: admin/spec/lib/validator/hdf-validation-spec.js (hunk 1 above)
Comment (jon-shipley) [reply admin/spec/lib/validator/hdf-validation-spec.js]: removed!

File: admin/services/jwt-service.js (hunk 2 above)
Comment (jon-shipley) [reply admin/services/jwt-service.js]: true - but I _like_ it this way.  The compiler will handle it.

**Comment by GeorgeChatzigiannis** [comment]
@jon-shipley conflicts

**Pull request merged**